homeassistant/components/homekit/type_covers.py

```diff
@@ -135,6 +174,21 @@ def update_state(self, new_state):
         else:
             self.char_position_state.set_value(2)
 
+        # update tilt
+        current_tilt = new_state.attributes.get(ATTR_CURRENT_TILT_POSITION)
+        if isinstance(current_tilt, (float, int)):
+            # HomeKit sends values between -90 and 90.
+            # We'll have to normalize to [0,100]
+            current_tilt = (current_tilt / 100.0 * 180.0) - 90.0
+            current_tilt = int(current_tilt)
+            self.char_current_tilt.set_value(current_tilt)
+
+            if self._homekit_target_tilt is None or abs(
+                current_tilt - self._homekit_target_tilt < 1
+            ):
+                self.char_target_tilt.set_value(current_tilt)
+                self._homekit_target = None
```

Comment:
Good catch, @bdraco! I poorly copy-pasted from the positioning code... 🙁

Tbh, I don't understand the reason why this conditional block is (still) needed in the first place. Couldn't this just be?
```suggestion
            self.char_target_tilt.set_value(current_tilt)
```

Is this to avoid syncing back the target_position if initiated by HomeKit? Because you already made sure that this is taken care of in [pyhap.accessory_driver:44](https://github.com/ikalchev/HAP-python/blob/06f73da41ba3a0cdf7f72857f7cd68dc6115667d/pyhap/accessory_driver.py#L454). If so, we could get rid of the `_homekit_target*` instance attributes altogether.

Other than that, the `abs(diff)` piece may make sense if it is to "snap" to the values in the end.

Reply:
Its been in there since @cdce8p added the original support.  It looks like its doing this to normalize the cover position since all covers aren't going to support the precision that homekit is so the state will appear out of sync